### PR TITLE
display test cases passing percentage

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-4697](https://jira.cms.gov/browse/MAT-4697)
(Optional) Related Tickets:

### Summary
display test cases passing percentage on test case list page
### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/api/CalculationService.ts [hunk 1]
```diff
@@ -185,6 +185,19 @@ export class CalculationService {
     return testCaseResultMap;
   }
 
+  getPassingPercentageForTestCases(testCases: TestCase[]) {
+    const totalTestCases = testCases?.length;
+    const passedTests = testCases?.filter(
+      (testCaseNow) => testCaseNow.executionStatus === "pass"
```
Comment (adongare) [inline src/api/CalculationService.ts]: this can be just `testCase`

File: src/api/CalculationService.ts [hunk 2]
```diff
@@ -185,6 +185,19 @@ export class CalculationService {
...
+    ).length;
+
+    const testCasesPassingDetails = {
+      passPercentage: Math.floor((passedTests / totalTestCases) * 100),
+      passFailRatio: passedTests + "/" + totalTestCases,
+    };
+    return testCasesPassingDetails;
```
Comment (adongare) [inline src/api/CalculationService.ts]: no need of this. You can return the result object directly.

File: src/api/CalculationService.ts (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/api/CalculationService.ts]: thank you, I have changed it

File: src/api/CalculationService.ts (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/api/CalculationService.ts]: sure, I have changed it

File: src/components/testCaseList/TestCaseList.tsx [hunk 3]
```diff
@@ -36,6 +41,11 @@ const TestCaseList = () => {
   const [activeTab, setActiveTab] = useState<string>("passing");
   const [executeAllTestCases, setExecuteAllTestCases] =
     useState<boolean>(false);
+  const [passingCoverage, setPassingCoverage] =
```
Comment (adongare) [inline src/components/testCaseList/TestCaseList.tsx]: this is not really coverage. maybe testCaseExecutionStats? passFailStats or something else

Comment (sb-prateekkeerthi) [reply src/components/testCaseList/TestCaseList.tsx]: Sure i'll change it

**Review by adongare — APPROVED** [review]

**Pull request merged**